main.tf

```
@@ -244,63 +240,39 @@ resource "google_storage_bucket" "gitlab-runner-cache" {
   location = "${var.region}"
 }
 // GKE Cluster
-resource "google_container_cluster" "gitlab" {
-  project            = "${var.project_id}"
-  name               = "gitlab"
-  location           = "${var.region}"
-  min_master_version = "${var.gke_version}"
-
-  # We can't create a cluster with no node pool defined, but we want to only use
-  # separately managed node pools. So we create the smallest possible default
-  # node pool and immediately delete it.
-  remove_default_node_pool = true
-
-  initial_node_count = 1
-
-  network    = "${google_compute_network.gitlab.self_link}"
-  subnetwork = "${google_compute_subnetwork.subnetwork.self_link}"
-
-  ip_allocation_policy {
-    cluster_secondary_range_name  = "gitlab-cluster-pod-cidr"
-    services_secondary_range_name = "gitlab-cluster-service-cidr"
-  }
-
-  enable_legacy_abac = true
+module "gke" {
+  source = "terraform-google-modules/kubernetes-engine/google"
+  version = "~> 9.0"
 
-  # Setting an empty username and password explicitly disables basic auth
-  master_auth {
-    username = ""
-    password = ""
+  # Create an implicit dependency on service activation
+  project_id                 = google_project_service.gke.project
 
-    client_certificate_config {
-      issue_client_certificate = true
-    }
-  }
+  name                       = "gitlab"
+  region                     = var.region
+  regional = true
+  kubernetes_version = var.gke_version
 
-  node_config {
-    oauth_scopes = [
-      "https://www.googleapis.com/auth/compute",
-      "https://www.googleapis.com/auth/devstorage.read_only",
-      "https://www.googleapis.com/auth/logging.write",
-      "https://www.googleapis.com/auth/monitoring",
-    ]
-  }
+  remove_default_node_pool=true
+  initial_node_count = 1
```

Not sure if we need to change this one and the Node pool one...
```suggestion
  initial_node_count = 3
```